src/formatDate.js

```diff
@@ -8,7 +8,71 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const oldDate = date.split(fromFormat[fromFormat.length - 1]);
+  const newDate = [];
+  let day = '';
+  let month = '';
+  let year = '';
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    switch (fromFormat[i]) {
+      case 'DD': {
+        day = oldDate[i];
+
+        break;
+      }
+
+      case 'MM': {
+        month = oldDate[i];
+
+        break;
+      }
+
+      default: {
+        year = oldDate[i];
+
+        break;
+      }
+    }
+  }
```

Comment:
It's good that you're using a loop to process the `fromFormat` array, but the code could be made more concise and readable by using an object to map the `fromFormat` elements to the corresponding parts of the `oldDate` array. This would eliminate the need for the `day`, `month`, and `year` variables and the switch statement.